src/cache.rs

```diff
@@ -21,6 +23,30 @@ pub struct Cache {
     os: OsType,
 }
 
+#[derive(Debug)]
```

Comment:
I agree for public types in libraries, but for internally used types, I usually only derive the types that are actually used, because deriving unused types gives more work to the optimizer (=longer compilation times) and has the potential of making the binary bigger (if the optimizer isn't good enough). So in this case, I'd not add more derives. `Debug` is definitely good though, it's required for some error handling cases!

Reply:
So I'm not going to derive the other traits (until needed since this is not a public library to help reduce compilation times.